src/ssl.c

```diff
@@ -5491,11 +5517,19 @@ static int ProcessBufferTryDecode(WOLFSSL_CTX* ctx, WOLFSSL* ssl, DerBuffer* der
                     }
 
                     *keyFormat = ED25519k;
-                    if (ssl && ssl->options.side == WOLFSSL_SERVER_END) {
-                        *resetSuites = 1;
+                    if (ssl != NULL) {
+                        /* ED25519 requires caching enabled for tracking message 
```

Comment:
@SparkiDev , I found an edge case where on the client side, not loading the private key till after the `wolfSSL_new` was causing the `options.cacheMessages` to not get set, so `EdDSA_Update` was not being run to calculate the message digest used by the Ed25519Sign, so the input was `NULL` to the sign, which produced a BAD_FUNC_ARG. The new API unit tests for crypto callbacks test this now with `./configure --enable-ed25519 --enable-curve25519 --enable-cryptocb`.

Reply:
Yep. Always testing with key set against context.
Thanks!